### PR TITLE
Initial release

### LINKED ISSUE #40: npm install shx
.. doesn't work. Nor does npm install shelljs expose shx.

How do I do an npm install -D of shx?

### PR DESCRIPTION
**Pull request by levithomason**
This PR will conclude any open issues.  Folks are ready to use it (#29 #40) and I believe we ready to let them start.

I've added release scripts that are battle tested and a README snippet on how to use them.  Once this PR is ready, we'll just have to run this to release the first `v0.1.0` to NPM:

```shell
npm run release:minor
```

- [ ] add npm owner `nfischer`
- [ ] add npm owner `ariporad`

Closes #40.

### CODE REVIEW

File: scripts/release.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
```
Comment (levithomason) [inline scripts/release.sh]: You can see more about this script on the [watch](https://github.com/mikeal/watch/pull/104) repo.  We also use it at my day job for all our apps and libraries, via our [`ta-scripts`](https://github.com/TechnologyAdvice/ta-scripts) package.

So, it's battle tested and works great.

Comment (nfischer) [inline scripts/release.sh]: @levithomason there are no licensing issues, right? Just wanted to be sure that we have permission to copy that script into our code-base without also copying in its Apache license.

**Comment by levithomason** [comment]
It is copied form TA.  All open source at TA is MIT.  Though, I just updated the [`ta-scripts` license](https://github.com/TechnologyAdvice/ta-scripts/blob/master/package.json#L20) to reflect that.  All clear.

**Comment by levithomason** [comment]
To be clear, I first wrote the script for TA, then copied to watch, now to shx.

**Comment by nfischer** [comment]
Ah, thanks for the update. Sounds good to me :+1:

File: README.md [hunk 2]
```diff
@@ -58,6 +58,18 @@ However, `shx` is ideal for one-liners inside `package.json`:
 }
 ```
 
+## Contributing
+
+### Releasing
```
Comment (ariporad) [inline README.md]: I think this should be in `CONTRIBUTING.md`, it's not useful to most README users.

**Comment by ariporad** [comment]
One small nit, otherwise this LGTM.

One thing though, I feel like the first version should be `0.0.1-rc1` (or `0.0.1-beta`), this way we can get some feedback first.

Thoughts?

**Comment by nfischer** [comment]
I don't care as much about the version number. I'm fine with 0.0.1, 0.1.0, or some sort of rc1

File: README.md (hunk 2 above)
Comment (levithomason) [inline README.md]: Agreed, updating.

**Comment by levithomason** [comment]
0.1.0 is already not a real version, see the semver [FAQ](http://semver.org/#how-should-i-deal-with-revisions-in-the-0yz-initial-development-phase) regarding this and when to release [1.0.0](http://semver.org/#how-do-i-know-when-to-release-100).  If we were already on a major, then we'd want to possibly field an rc.

**Comment by nfischer** [comment]
Why is this failing CI?

**Comment by levithomason** [comment]
Hm, I had similar issues earlier today with babel:

```shell
> shx@0.0.1 prepublish /Users/travis/build/shelljs/shx
> npm run build

> shx@0.0.1 prebuild /Users/travis/build/shelljs/shx
> rimraf lib

> shx@0.0.1 build /Users/travis/build/shelljs/shx
> babel src -d lib

TypeError: undefined is not iterable!
```

I'm wondering if babel shipped something that not longer allows building into dirs that don't exist?  Errors on other builds today were about an undefined path.

I'm gonna push a few attempts at fixing this.

**Comment by levithomason** [comment]
A rebuild seems to have worked for the push build.  Just restarted the PR build to see.  This could be due to rebase / force push I did soon after the first build, actually.

**Comment by levithomason** [comment]
The fails are not related to the changes here, merging.

**Pull request merged**